### PR TITLE
8346324: javax/swing/JScrollBar/4865918/bug4865918.java fails in CI

### PR DESCRIPTION
**Pull request by prsadhuk**
javax/swing/JScrollBar/4865918/bug4865918.java fails in CI citing

java.lang.RuntimeException: The scrollbar block increment is incorrect
at bug4865918.main(bug4865918.java:60) 

Seems like scrollbar is not in focus when mouse is pressed..Used CountDownLatch to mage focus gain more determinisitic.
Fix is passing in CI in several CI systems

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8346324](https://bugs.openjdk.org/browse/JDK-8346324): javax/swing/JScrollBar/4865918/bug4865918.java fails in CI (**Bug** - P4)

### Reviewers
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - **Reviewer**) Review applies to [2eb13c95](https://git.openjdk.org/jdk/pull/22783/files/2eb13c950fdf9f7683bb02ac371a0bf79d7bf68d)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22783/head:pull/22783` \
`$ git checkout pull/22783`

Update a local copy of the PR: \
`$ git checkout pull/22783` \
`$ git pull https://git.openjdk.org/jdk.git pull/22783/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22783`

View PR using the GUI difftool: \
`$ git pr show -t 22783`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22783.diff">https://git.openjdk.org/jdk/pull/22783.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22783#issuecomment-2547720794)
</details>

### CODE REVIEW

File: test/jdk/javax/swing/JScrollBar/4865918/bug4865918.java [hunk 1]
```diff
@@ -34,16 +34,20 @@
...
 import java.util.Date;
 
 public class bug4865918 {
 
     private static TestScrollBar sbar;
     private static JFrame frame;
+    private static volatile CountDownLatch latch;
 
     public static void main(String[] argv) throws Exception {
+        latch = new CountDownLatch(1);
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/JScrollBar/4865918/bug4865918.java:50]: Initialise the latch at its declaration.
```suggestion
    private static final CountDownLatch latch = new CountDownLatch(1);

    public static void main(String[] argv) throws Exception {
```

`mousePressLatch` would be a more descriptive name.

File: test/jdk/javax/swing/JScrollBar/4865918/bug4865918.java [hunk 2]
```diff
@@ -52,6 +56,7 @@ public static void main(String[] argv) throws Exception {
             robot.delay(1000);
 
             SwingUtilities.invokeAndWait(() -> sbar.pressMouse());
+            latch.await();
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/JScrollBar/4865918/bug4865918.java:59]: You should use timed `await` and throw an exception if the wait times out. This way the timeout would be handled more gracefully… and more quickly, a two-second timeout should be enough.

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]
In the description you say, “scrollbar is not in focus when mouse is pressed.” Yet there's nothing in the test which handles the focus… even in the updated test.

However, I agree that your update makes the test more reliable, the latch guarantees the synthetic mouse event is handled before analysing the result.

So, it looks good exception for a couple of nits.

File: test/jdk/javax/swing/JScrollBar/4865918/bug4865918.java [hunk 3]
```diff
@@ -34,14 +34,18 @@
 import javax.swing.JFrame;
 import javax.swing.JScrollBar;
 import javax.swing.SwingUtilities;
+import java.awt.event.MouseAdapter;
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JScrollBar/4865918/bug4865918.java:37]: awt imports can be grouped together.

**Review by kumarabhi006 — COMMENTED** [review]
Seems good.

**Review by kumarabhi006 — APPROVED** [review]

File: test/jdk/javax/swing/JScrollBar/4865918/bug4865918.java [hunk 4]
```diff
@@ -56,7 +56,7 @@ public static void main(String[] argv) throws Exception {
             robot.delay(1000);
 
             SwingUtilities.invokeAndWait(() -> sbar.pressMouse());
-            latch.await();
+            mousePressLatch.await(2, TimeUnit.SECONDS);
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/JScrollBar/4865918/bug4865918.java:59]:
```suggestion
            if (!mousePressLatch.await(2, TimeUnit.SECONDS)) {
                throw new RuntimeException("Timed out waiting for mouse press");
            }
```

If you don't throw the exception, the test will continue to run… Yet we're sure it is likely to fail.

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

File: test/jdk/javax/swing/JScrollBar/4865918/bug4865918.java (hunk 4 above)
Comment (prsadhuk) [reply test/jdk/javax/swing/JScrollBar/4865918/bug4865918.java]: ok..modified..

**Review by aivanov-jdk — APPROVED** [review]

**Pull request closed without merging**